### PR TITLE
SONAR-8857 make ws/qualityprofiles/delete organization aware

### PR DESCRIPTION
**Pull request by bartfastiel**
This PR is based on #1783.

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java [hunk 1]
```diff
@@ -109,10 +109,12 @@ public QualityProfileDto getProfile(DbSession dbSession, QProfileReference ref)
     QualityProfileDto profile;
     if (ref.hasKey()) {
       profile = dbClient.qualityProfileDao().selectByKey(dbSession, ref.getKey());
+      checkFound(profile, "Quality Profile with key '"+ref.getKey()+"' does not exist");
     } else {
       OrganizationDto org = getOrganizationByKey(dbSession, ref.getOrganizationKey().orElse(null));
       profile = dbClient.qualityProfileDao().selectByNameAndLanguage(org, ref.getName(), ref.getLanguage(), dbSession);
+      checkFound(profile, "Quality Profile for language '"+ref.getLanguage()+"' and name '"+ref.getName()+"' does not exist"+ref.getOrganizationKey().map(o ->" in organization '"+o+"'").orElse(""));
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java]: MINOR Split this 198 characters long line (which is greater than 180 authorized). [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00103)

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/DeleteAction.java [hunk 2]
```diff
@@ -26,21 +26,28 @@
 import org.sonar.api.server.ws.WebService.NewController;
 import org.sonar.db.DbClient;
 import org.sonar.db.DbSession;
+import org.sonar.db.organization.OrganizationDto;
 import org.sonar.db.qualityprofile.QualityProfileDto;
 import org.sonar.server.qualityprofile.QProfileFactory;
 import org.sonar.server.qualityprofile.QProfileRef;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/DeleteAction.java]: MINOR Remove this unused import 'org.sonar.server.qualityprofile.QProfileRef'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/DeleteAction.java [hunk 3]
```diff
@@ -26,21 +26,28 @@
 import org.sonar.api.server.ws.WebService.NewController;
 import org.sonar.db.DbClient;
 import org.sonar.db.DbSession;
+import org.sonar.db.organization.OrganizationDto;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/DeleteAction.java]: MINOR Remove this unused import 'org.sonar.db.organization.OrganizationDto'. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 6 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 3 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### 3 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [QProfileReference.java#L120](https://github.com/SonarSource/sonarqube/blob/5e1bc5978d2124a578a6114e48930829b53ce577/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileReference.java#L120): Reduce the number of returns of this method 6, down to the maximum allowed 5. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1142)
1. ![INFO][INFO] [QProfileWsSupport.java#L79](https://github.com/SonarSource/sonarqube/blob/5e1bc5978d2124a578a6114e48930829b53ce577/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java#L79): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QProfileWsSupport.java#L98](https://github.com/SonarSource/sonarqube/blob/5e1bc5978d2124a578a6114e48930829b53ce577/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java#L98): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues
* ![MINOR][MINOR] 3 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [QProfileWsSupport.java#L79](https://github.com/SonarSource/sonarqube/blob/3c69333e0ce628e83b1e5cf2bd426c96cbee1bd6/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java#L79): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QProfileWsSupport.java#L98](https://github.com/SonarSource/sonarqube/blob/3c69333e0ce628e83b1e5cf2bd426c96cbee1bd6/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java#L98): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/DeleteAction.java (hunk 3 above)
Comment (sns-seb) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/DeleteAction.java]: @SonarTech has a valid point

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java (hunk 1 above)
Comment (sns-seb) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java]: `checkFound` accepts patterns with '%s' (`String.format` style), don't use String concatenation (it has a cost even if check doesn't fail)

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java [hunk 4]
```diff
@@ -109,10 +109,12 @@ public QualityProfileDto getProfile(DbSession dbSession, QProfileReference ref)
     QualityProfileDto profile;
     if (ref.hasKey()) {
       profile = dbClient.qualityProfileDao().selectByKey(dbSession, ref.getKey());
+      checkFound(profile, "Quality Profile with key '"+ref.getKey()+"' does not exist");
```
Comment (sns-seb) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java]: `checkFound` accepts patterns with '%s' (`String.format` style)

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/DeleteActionTest.java [hunk 5]
```diff
@@ -116,89 +122,176 @@ public void delete_nominal_with_language_and_name() throws Exception {
...
+      .execute();
   }
 
   @Test
   public void fail_on_missing_arguments() throws Exception {
     logInAsQProfileAdministrator();
 
     expectedException.expect(IllegalArgumentException.class);
-    expectedException.expectMessage("Both profile language and name must be set");
+    expectedException.expectMessage("If no quality profile key is specified, language and name must be set");
```
Comment (sns-seb) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/DeleteActionTest.java]: "**both** language and name must be set"?

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/DeleteActionTest.java [hunk 6]
```diff
@@ -116,89 +122,176 @@ public void delete_nominal_with_language_and_name() throws Exception {
...
+      .execute();
   }
 
   @Test
   public void fail_on_missing_language() throws Exception {
     logInAsQProfileAdministrator();
 
     expectedException.expect(IllegalArgumentException.class);
-    expectedException.expectMessage("Both profile language and name must be set");
+    expectedException.expectMessage("If no quality profile key is specified, language and name must be set");
```
Comment (sns-seb) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/DeleteActionTest.java]: "**both** language and name must be set"?

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/DeleteActionTest.java [hunk 7]
```diff
@@ -116,89 +122,176 @@ public void delete_nominal_with_language_and_name() throws Exception {
...
   @Test
   public void fail_on_too_many_arguments() throws Exception {
     logInAsQProfileAdministrator();
+    QualityProfileDto profile = insertQualityProfile();
 
     expectedException.expect(IllegalArgumentException.class);
-    expectedException.expectMessage("Either key or couple language/name must be set");
-
-    tester.newPostRequest("api/qualityprofiles", "delete").setParam("profileName", "Polop").setParam("language", xoo1.getKey()).setParam("profileKey", "polop").execute();
+    expectedException.expectMessage("When providing a quality profile key, neither of organization/language/name must be set");
```
Comment (sns-seb) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/DeleteActionTest.java]: "neither organization nor language nor name must be set"

or "none of organization, language or name must be set"

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/DeleteAction.java (hunk 3 above)
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/DeleteAction.java]: Done.

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java (hunk 4 above)
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java]: Done.

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java (hunk 1 above)
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java]: Done.

Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java]: Done.

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/DeleteActionTest.java (hunk 6 above)
Comment (bartfastiel) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/DeleteActionTest.java]: Not needed necessarily imho.

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/DeleteActionTest.java (hunk 7 above)
Comment (bartfastiel) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/DeleteActionTest.java]: I agree, that this text would have been better, but I would like to keep it for the moment, for not to have to change the text everywhere.

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [QProfileWsSupport.java#L79](https://github.com/SonarSource/sonarqube/blob/5c78167986fb00a589dcdf7024a754f3864db8f4/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java#L79): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. ![INFO][INFO] [QProfileWsSupport.java#L98](https://github.com/SonarSource/sonarqube/blob/5c78167986fb00a589dcdf7024a754f3864db8f4/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileWsSupport.java#L98): Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'

**Pull request merged**